install-panel.sh

```diff
@@ -711,49 +682,131 @@ function main {
+  #When you want to call the total UFW function, you call this function
+
+  read -p "Are you going to be installing the Daemon on this machine (y/N): " DaemonAsk
+
+  #If you are going to be installing the daemon and you want a firewall, by default the 
+  #ports 80, 22, 2022 and 8080 Are allowed through the firewall
+  if [[ "$DaemonAsk" =~ [Yy] ]]; then
+    read -p "Do you want to enable a firewall? (y/N): " ConfirmFirewall
+    if [[ "$ConfirmFirewall" =~ [Yy] ]]; then
+      apt install ufw -y
+      echo -e "\nEnabling Uncomplicated Firewall (UFW)"
+      echo " This is to do the inital setup of UFW, opening the following ports."
+      echo " If you're running SSL you need to add the port 443"
+      echo -e "  80\n  22\n  2022\n  8080"
+      read -p "Would you like to add additional ports besides the ones listed? (y/N): " ConfirmAdditional
+
+      #This is where it asks the user if they want to add any additional ports
+      if [[ "$ConfirmAdditional" =~ [Yy] ]]; then
+        ufwAdditional-Ubuntu
+        ufw allow 80 | grep 'zzz'
+        ufw allow 22 | grep 'zzz'
+        ufw allow 2022 | grep 'zzz'
+        ufw allow 8080 | grep 'zzz'
+        ufw enable
+        ufw status numbered | sed '/v6/d'
+        sleep 3
+      fi
+    fi
+  else
+
+    #If you aren't going to be installing the daemon and you want a firewall, only the
+    #ports 22 and 80 are opened through ufw.
+    read -p "Do you want to enable a firewall? (y/N): " ConfirmFirewall
+    if [[ "$ConfirmFirewall" =~ [Yy] ]]; then
+      apt install ufw -y
+      echo -e "\nEnabling Uncomplicated Firewall (UFW)"
+      echo " This is to do the inital setup of UFW, opening the following ports."
+      echo " If you're running SSL you need to add the port 443"
+      echo -e "  80\n  22"
+      read -p "Would you like to add additional ports besides the ones listed? (y/N): " ConfirmAdditional
+
+      #This is where it asks the user if they want to add any additional ports
+      if [[ "$ConfirmAdditional" =~ [Yy] ]]; then
+        ufwAdditional-Ubuntu
+        ufw allow 80 | grep 'zzz'
+        ufw allow 22 | grep 'zzz'
+        ufw enable
+        ufw status numbered | sed '/v6/d'
+        sleep 3
+      fi  
+    fi
+  fi
+}
 
+function ufwAdditional-Ubuntu {
+  #This loops, asking the user what ports they want within the firewall, then adds the ports.
+  #User can then type 'done' to exit.
+  while true
+    do
+      read -p " Enter the port you want to open (type 'done' to exit) 0-65535: " Port
+      if [[ ! "$Port" =~ [Dd][Oo][Nn][Ee] ]]; then
+        ufw allow $Port | grep 'zzz'
+      else
+        break
+      fi
+    done
+}
+
+function autoCertUbuntu18 {
+  #Set up the certificate with LetsEncrypt to use HTTPS
+  echo -e "\nMake sure you choose Option 1, and create a Standalone Web Server during the certificate"
+  read -p "* Enter your FQDN panel.example.com: " fqdn
+  apt install certbot -y
+  certbot certonly -d $fqdn
```

Comment:
Why are you using `certonly`? Certbot has the `nginx` plugin which should work for this.